### PR TITLE
Identify and handle runtime values when generating displayed text

### PR DESCRIPTION
**Pull request by guerler**
This PR fixes and issue related to runtime values not being properly handled when generating the user facing display text value.

### CODE REVIEW

File: lib/galaxy/tools/parameters/basic.py [hunk 1]
```diff
@@ -156,6 +155,11 @@ def value_from_basic( self, value, app, ignore_errors=False ):
...
 
     def value_to_display_text( self, value, app=None ):
+        if is_runtime_value( value ):
+            return "Not available."
+        return self.to_text( value, app )
+
+    def to_text( self, value, app ):
         """
         Convert a value to a text representation suitable for displaying to
         the user
```
Comment (nsoranzo) [inline lib/galaxy/tools/parameters/basic.py]: The following doctests should be updated to call `to_text()` or moved to the `value_to_display_text()` method above.

File: lib/galaxy/tools/parameters/basic.py [hunk 2]
```diff
@@ -156,6 +155,11 @@ def value_from_basic( self, value, app, ignore_errors=False ):
             return self.to_python( value, app )
 
     def value_to_display_text( self, value, app=None ):
+        if is_runtime_value( value ):
+            return "Not available."
+        return self.to_text( value, app )
+
+    def to_text( self, value, app ):
```
Comment (nsoranzo) [inline lib/galaxy/tools/parameters/basic.py]: You can remove `app` from the list of paramaters, it's not used either here or in the overridden methods.

File: lib/galaxy/tools/parameters/basic.py (hunk 1 above)
Comment (guerler) [reply lib/galaxy/tools/parameters/basic.py]: Excellent. Thanks.

**Comment by guerler** [comment]
@galaxybot test this

**Review by nsoranzo — APPROVED** [review]

**Comment by bgruening** [comment]
Thanks Sam!

**Pull request merged**